daemon/logger/syslog/syslog.go

```diff
@@ -25,9 +28,40 @@ func init() {
 	}
 }
 
+func parseConnectionUrl(c map[string]string) (string, string, error) {
+	rawurl, _ := c["syslog-connection-url"]
+	if urlutil.IsURL(rawurl) {
+		url, err := url.Parse(rawurl)
+		if err != nil {
+			return "", "", err
+		}
+
+		proto := url.Scheme
+		address := url.Host
+		if strings.Count(address, ":") == 0 {
+			address = fmt.Sprintf("%s:514", address)
```

Comment:
Well, you are already doing that, but SplitHostPort is a bit more sophisticated and works with ipv6.